File: websites/jobs/web2/src/services/position.service.js
import http from "../http-common";

class PositionService {
    getAll() {
      return http.get("/position");
    }
  
    get(id) {
      return http.get(`/position/${id}`);
    }
  
    create(data) {
      return http.post("/position", data);
    }
  
    update(id, data) {
      return http.put(`/position/${id}`, data);
    }
  
    delete(id) {
      return http.delete(`/position/${id}`);
    }
  
    deleteAll() {
      return http.delete(`/position`);
    }
  
    findByTitle(title) {
      return http.get(`/apply?title=${title}`);
    }
  }
  
  export default new PositionService();

File: websites/jobs/jobsapply/src/main/java/club/snp/jobsapply/service/PositionService.java
package club.snp.jobsapply.service;

import club.snp.jobsapply.entity.Apply;
import club.snp.jobsapply.entity.Position;
import club.snp.jobsapply.repository.PositionRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class PositionService {
    PositionRepository positionRepository;

    PositionService(PositionRepository positionRepository) {
        this.positionRepository = positionRepository;
    }

    public List<Position> getAll() {
        return positionRepository.findAll();
    }

    public void deleteById(long id) {
        positionRepository.deleteById(id);
    }

    public Optional<Position> findById(long id) {
        Optional<Position> positionData = positionRepository.findById(id);
        return positionData;
    }

    public Position save(Position position) {
        positionRepository.save(position);
        return position;
    }

    public Optional<Position> findByTitle(String title) {
        Optional<Position> positionData = positionRepository.findByTitle(title);
        return positionData;
    }
}
